beard/metrics/clustering.py

```diff
@@ -171,4 +168,36 @@ def _group_samples_by_cluster_id(labels):
     """
     groupped_samples = groupby(np.argsort(labels), lambda i: labels[i])
 
-    return (list(group) for _, group in groupped_samples)
+    return {k: list(values) for k, values in groupped_samples}
+
+
+def merge_distance(R_flat, S_flat, fs=lambda x, y: 1, fm=lambda x, y: 1):
```

Comment:
Make this a private function as well, and call it `general_merge_distance` to match with algorithm original name.